collectfast/strategies/boto3.py

```diff
@@ -29,6 +49,51 @@ def _clean_hash(quoted_hash: Optional[str]) -> Optional[str]:
         assert quoted_hash[0] == quoted_hash[-1] == '"'
         return quoted_hash[1:-1]
 
+    def get_aws_hash(self, open_handle: IO) -> str:
+        """Calculate multipart-friendly hash using `multipart_chunksize` chunk size."""
+        chunk_hashes = []
+        while True:
+            data = open_handle.read(multipart_chunksize)
+            if not data:
+                break
+            chunk_hashes.append(hashlib.md5(data))
+
+        if len(chunk_hashes) < 1:
+            return hashlib.md5().hexdigest()
+
+        if len(chunk_hashes) == 1:
+            return chunk_hashes[0].hexdigest()
+
+        digests = b"".join(m.digest() for m in chunk_hashes)
+        digests_md5 = hashlib.md5(digests)
+        return "{}-{}".format(digests_md5.hexdigest(), len(chunk_hashes))
+
+    def get_gzipped_aws_hash(self, open_handle: IO) -> str:
+        """Calculate multipart-friendly gzipped hash."""
+        buffer = BytesIO()
+        zf = gzip.GzipFile(mode="wb", fileobj=buffer, mtime=0.0)
+        zf.write(force_bytes(open_handle.read()))
```

Comment:
Can we eliminate this `force_bytes()` call? It seems like we're already reading bytes from the filesystem in `get_local_file_hash()`, so we could probably just limit the type of `open_handle` to `io.BytesIO`?

It might be worth trying to switch to text-based write and read modes but that is probably out of scope for this PR :)

Reply:
Good call.